src/Lib.P2P/Routing/DhtService.cs

```diff
@@ -0,0 +1,444 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Logging;
+using MultiFormats;
+using ProtoBuf;
+using Semver;
+
+namespace Lib.P2P.Routing
+{
+    /// <summary>
+    ///   DHT Protocol version 1.0
+    /// </summary>
+    public class DhtService : IDhtService
+    {
+        private static ILog _log = LogManager.GetLogger(typeof(DhtService));
+
+        /// <inheritdoc />
+        public virtual string Name { get; } = "libp2p-cs/kad";
+
+        /// <inheritdoc />
+        public SemVersion Version { get; } = new SemVersion(1);
+
+        /// <summary>
+        ///   Provides access to other peers.
+        /// </summary>
+        public ISwarmService SwarmService { get; set; }
+
+        /// <summary>
+        ///  Routing information on peers.
+        /// </summary>
+        public RoutingTable RoutingTable;
+
+        /// <summary>
+        ///   Peers that can provide some content.
+        /// </summary>
+        public ContentRouter ContentRouter;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)